evap/contributor/views.py

```diff
@@ -130,13 +129,16 @@ def course_edit(request, course_id):
     formset = InlineContributionFormset(request.POST or None, instance=course, can_change_responsible=False, form_kwargs={'course': course})
 
     forms_are_valid = course_form.is_valid() and formset.is_valid()
-
     if forms_are_valid and not preview:
         if post_operation not in ('save', 'approve'):
             raise SuspiciousOperation("Invalid POST operation")
 
-        course_form.save(user=user)
-        formset.save()
+        if formset.has_changed():
+            formset.save()
+            # Save form, even if only formset has changed, to update last_modified_user
+            course_form.save(user=request.user)
+        elif course_form.has_changed():
+            course_form.save(user=request.user)
 
         if post_operation == 'approve':
```

Comment:
since you're already on this: maybe you could also improve the following lines so when approving the course without any further changes it shows a message `"Successfully approved course."`? we would then have three different messages for update / approve / update and approve. it would be less misleading than the current message.

Reply:
done!